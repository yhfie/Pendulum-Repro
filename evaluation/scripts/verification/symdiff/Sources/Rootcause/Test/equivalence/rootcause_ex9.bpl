// func_Eq.bpl
//
// Example with equivalence checking 
//

const unique a:int;

function f(h:[int]int, i:int): bool;
function fa(h:[int]int, i:int): [int]int;
function fb(h:[int]int, i:int): [int]int;

procedure EQ_A(index:int, tmp:int, h:[int]int) {

  //Version 1
  var t1:bool, i1:int;
  var t2:bool, i2:int;
  var h1:[int]int, h2:[int]int;

  t1 := f(h, index);
  
  if (t1) {
       h1 := fa(h, index);
  } else {
       h1 := fb(h, index);
  }	

  //this is the marker that is understood by Rootcause
  goto AA_INSTR_EQ_BODY$1;
  AA_INSTR_EQ_BODY$1: 
   
  //Version 2

  t2 := f(h, index + 1);
  
  if (t2) {
       h2 := fa(h, index);
  } else {
       h2 := fb(h, index);
  }	


  assert (h1 == h2);
}
