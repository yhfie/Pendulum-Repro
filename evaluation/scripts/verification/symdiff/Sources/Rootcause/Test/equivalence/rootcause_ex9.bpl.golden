BENCHMARK:rootcause_ex9.bpl(ERRORFREE)
Processing file rootcause_ex9.bpl
############# Implementation = EQ_A #################
Cause ==>
leftAssignment: t1 := f(h, index);
rightAssignment: t2 := f(h, index + 1);
